tests/unittest/_torch/attention/kv_cache_utils.py

```
@@ -0,0 +1,142 @@
+    num_cached_tokens: Sequence[int],
+    *,
+    device: str = "cuda",
+) -> torch.Tensor:
+    """Build per-token RoPE position ids for a packed (varlen) batch.
+
+    For sequence ``i`` the query tokens occupy absolute positions
+    ``[cached_i, cached_i + seq_len_i)``. Returns a 1-D int tensor of shape
+    ``[sum(seq_lens)]`` matching the packed q layout used by the backends.
+    """
+    pieces = []
+    for s_len, cached in zip(seq_lens, num_cached_tokens):
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)